src/game.cpp

```diff
@@ -4378,10 +4360,21 @@ void Game::updateFrame(ProfilerGraph *graph, RunStats *stats, f32 dtime,
 	TimeTaker tt_draw("mainloop: draw");
 	driver->beginScene(true, true, skycolor);
 
+	if (runData.disable_fog) {
+		driver->getOverrideMaterial().reset();
+		driver->getOverrideMaterial().EnableFlags = video::EMF_FOG_ENABLE;
+		driver->getOverrideMaterial().EnablePasses = irr::scene::ESNRP_SKY_BOX
+				| irr::scene::ESNRP_SOLID | irr::scene::ESNRP_TRANSPARENT
+				| irr::scene::ESNRP_TRANSPARENT_EFFECT | irr::scene::ESNRP_SHADOW;
+	}
```

Comment:
Shouldn't lines bellow moved in else condition here?

Reply:
`driver->getOverrideMaterial().EnableFlags = 0;`? No. It is there to restore `OverrideMaterial` to a clean state.

It could be hidden under an equivalent `if` condition, though. Or replaced with `driver->getOverrideMaterial().reset();`. But whatever way is chosen, it remains fragile. `OverrideMaterial` simply doesn’t support stacking modifications.

Reply:
oops, anaglyph 3D mode will probably break this... it’s the only other place that uses `OverrideMaterial`

Reply:
Ah, thanks for the explanation.
No one uses 3D mode with this version of MultiCraft, so who the current patch is acceptable. Although for MC2 we should try not to break 3D.

Also, with the previous commit, using gles2 is impossible without MT shaders enabled, so maybe this patch is not required?